common/src/attestations/signers.rs

```diff
@@ -3,162 +3,53 @@
     dispute_manager: Address,
-}
-
-impl AttestationSigners {
-    pub fn new(
-        allocation_monitor: AllocationMonitor,
-        indexer_mnemonic: String,
-        chain_id: U256,
-        dispute_manager: Address,
-    ) -> Self {
-        let inner = Arc::new(AttestationSignersInner {
-            attestation_signers: Arc::new(RwLock::new(HashMap::new())),
-            allocation_monitor,
-            indexer_mnemonic,
-            chain_id,
-            dispute_manager,
-        });
-
-        let _update_loop_handle = {
-            let inner = inner.clone();
-            tokio::spawn(Self::update_loop(inner.clone()))
-        };
-
-        Self {
-            inner,
-            _update_loop_handle: Arc::new(_update_loop_handle),
-        }
-    }
-
-    pub async fn update_attestation_signers(inner: Arc<AttestationSignersInner>) {
-        let mut attestation_signers_write = inner.attestation_signers.write().await;
-        for allocation in inner
-            .allocation_monitor
-            .get_eligible_allocations()
-            .await
-            .values()
-        {
-            if let std::collections::hash_map::Entry::Vacant(e) =
-                attestation_signers_write.entry(allocation.id)
-            {
-                match attestation_signer_for_allocation(&inner.indexer_mnemonic, allocation)
-                    .and_then(|signer| {
-                        create_attestation_signer(
-                            inner.chain_id,
-                            inner.dispute_manager,
-                            signer,
-                            allocation.subgraph_deployment.id.bytes32(),
-                        )
-                    }) {
-                    Ok(signer) => {
-                        e.insert(signer);
-                        info!(
-                            "Found attestation signer for {{allocation: {}, deployment: {}}}",
-                            allocation.id,
-                            allocation.subgraph_deployment.id.ipfs_hash()
-                        );
-                    }
-                    Err(e) => {
-                        warn!(
-                        "Failed to find the attestation signer for {{allocation: {}, deployment: {}, createdAtEpoch: {}, err: {}}}",
-                        allocation.id, allocation.subgraph_deployment.id.ipfs_hash(), allocation.created_at_epoch, e
-                    )
-                    }
-                }
-            }
-        }
-    }
-
-    async fn update_loop(inner: Arc<AttestationSignersInner>) {
-        let mut watch_receiver = inner.allocation_monitor.subscribe();
+) -> Eventual<HashMap<Address, AttestationSigner>> {
+    // Keep a cache of the most recent 1000 signers around so we don't need to recreate them
+    // every time there is a small change in the allocations
+    let cache: &'static Mutex<LruCache<_, _>> = Box::leak(Box::new(Mutex::new(LruCache::new(
+        NonZeroUsize::new(1000).unwrap(),
+    ))));
```

Comment:
What is the rationale for using an LRU cache rather than updating a hashmap like before? It looks like a roundabout way of achieving a similar thing, but with the added hardcoded limit of 1000 allocations.